src/System.Threading.Tasks.Parallel/tests/ParallelForTests.cs

```diff
@@ -349,132 +262,349 @@ public static void TestParallelForEachConversions()
+
+            for (int i = inclusiveFrom; i < exclusiveTo; i++)
+            {
+                seqForIndices.Add(i);
+            }
+            
+            Parallel.For(inclusiveFrom, exclusiveTo, i =>
+            {
+                lock (parForIndices)
+                {
+                    parForIndices.Add(i);
+                }
+            });
+            parForIndices.Sort();
+            Assert.Equal(seqForIndices, parForIndices);
         }
 
-        [Fact]
-        public static void SequentialFor64ParityTest()
+        [Theory]
+        [InlineData(0, 100)]
+        [InlineData(-100, 100)]
+        [InlineData((long)int.MaxValue - 100, (long)int.MaxValue + 100)]
+        [InlineData(long.MaxValue - 1, long.MaxValue)]
+        // These fail for now.  Should be fixed when Huseyin implements new range-splitting logic.
+        //[InlineData(long.MaxValue - 100, long.MaxValue - 1)]
+        //[InlineData(long.MaxValue - 100, long.MaxValue - 2)]
+        //[InlineData(long.MaxValue - 100, long.MaxValue - 10)]
+        public static void SequentialFor64ParityTest(long inclusiveFrom, long exclusiveTo)
         {
-            SequentialFor64ParityTest(0, 100);
-            SequentialFor64ParityTest(-100, 100);
-            SequentialFor64ParityTest(-100, 100);
-            SequentialFor64ParityTest(-100, 100);
-            SequentialFor64ParityTest((long)Int32.MaxValue - 100, (long)Int32.MaxValue + 100);
-            SequentialFor64ParityTest((long)Int32.MaxValue - 100, (long)Int32.MaxValue + 100);
-            SequentialFor64ParityTest((long)Int32.MaxValue - 100, (long)Int32.MaxValue + 100);
-            SequentialFor64ParityTest(Int64.MaxValue - 1, Int64.MaxValue);
-            // These fail for now.  Should be fixed when Huseyin implements new range-splitting logic.
-            // SequentialFor64ParityTest(Int64.MaxValue - 100, Int64.MaxValue, 1);
-            // SequentialFor64ParityTest(Int64.MaxValue - 100, Int64.MaxValue, 2);
-            // SequentialFor64ParityTest(Int64.MaxValue - 100, Int64.MaxValue, 10);
+            var seqForIndices = new List<long>();
+            var parForIndices = new List<long>();
+
+            for (long i = inclusiveFrom; i < exclusiveTo; i++)
+            {
+                seqForIndices.Add(i);
+            }
+            
+            Parallel.For(inclusiveFrom, exclusiveTo, i =>
+            {
+                lock (parForIndices)
+                {
+                    parForIndices.Add(i);
+                }
+            });
+            parForIndices.Sort();
+
+            Assert.Equal(seqForIndices, parForIndices);
         }
 
-        [Fact]
-        public static void RunSimpleParallelForeachAddTest_Enumerable()
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(1024)]
+        [InlineData(1024 * 1024)]
+        // This one just stopped working around 07/08/08 -- Can cause a horrible, indecipherable crash.
```

Comment:
Let's get rid of this comment.  If there's actually a problem here, we'll find it and fix it.